gordo_components/server/views/anomaly.py

```diff
@@ -140,22 +139,21 @@ def _process_base_response(self, start_time: float = None):
         # To use this endpoint, we need a 'y' to calculate the errors.
         # It has either come from client providing it in POST or from
         # Influx during 'GET' part of `..common.extract_X_y` decorator
-        y = g.y
-        if y is None:
+        if g.y is None:
             message = {
                 "message": "Cannot perform anomaly without 'y' to compare against."
             }
             return make_response((jsonify(message), 400))
 
         # It is ok for y to be a subset of features in X, but we need at least one
         # to compare against to calculate an error.
-        if not any(col in y.columns for col in [t.name for t in self.tags]):
+        if not any(col in g.y.columns for col in [t.name for t in self.tags]):
             message = {"message": "y is not a subset of X, cannot do anomaly detection"}
             return make_response(jsonify(message), 400)
 
         # Now create an anomaly dataframe from the base response dataframe
         try:
-            anomaly_df = current_app.model.anomaly(g.data, y, frequency=self.frequency)
+            anomaly_df = current_app.model.anomaly(g.X, g.y, frequency=self.frequency)
         except AttributeError:
```

Comment:
Maybe we check for `AnomalyDetectorBase`; because an attribute error could arise within `.anomaly` and the resulting message wouldn't be true.

Reply:
Should be addressed in https://github.com/equinor/gordo-components/issues/393